pkg/controller/crd_annotator/controller.go

```diff
@@ -0,0 +1,325 @@
+	key := types.NamespacedName{
+		Name: common.ExternalSecretsObjectName,
+	}
+	if err := r.Get(ctx, key, es); err != nil {
+		if errors.IsNotFound(err) {
+			// NotFound errors, would mean the object hasn't been created yet and
+			// not required to reconcile yet.
+			r.log.V(1).Info("externalsecrets.openshift.operator.io object not found, skipping reconciliation", "key", key)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to fetch externalsecrets.openshift.operator.io %q during reconciliation: %w", key, err)
+	}
+
+	if common.IsInjectCertManagerAnnotationEnabled(es) {
+		return r.processReconcileRequest(es, req.NamespacedName)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// processReconcileRequest is the reconciliation handler to manage the resources.
+func (r *Reconciler) processReconcileRequest(es *operatorv1alpha1.ExternalSecrets, req types.NamespacedName) (ctrl.Result, error) {
+	var oErr error = nil
+	if req.Name == reconcileObjectIdentifier {
+		if err := r.updateAnnotationsInAllCRDs(); err != nil {
+			oErr = fmt.Errorf("failed while updating annotations in all CRDs: %w", err)
+		}
+	} else {
+		crd := &crdv1.CustomResourceDefinition{}
+		if err := r.Get(r.ctx, req, crd); err != nil {
+			// NotFound error will be ignored since CRDs are managed by OLM, and OLM will
+			// reconcile it.
+			if errors.IsNotFound(err) {
+				r.log.V(1).Info("crd managed by OLM is not found, skipping reconciliation", "crd", req)
+				return ctrl.Result{}, nil
+			}
+			oErr = fmt.Errorf("failed to fetch customresourcedefinitions.apiextensions.k8s.io %q during reconciliation: %w", req, err)
+		}
+		if err := r.updateAnnotations(crd); err != nil {
+			oErr = fmt.Errorf("failed to update annotations in %q: %w", req, err)
+		}
+	}
+
+	if err := r.updateCondition(es, oErr); err != nil {
+		return ctrl.Result{}, utilerrors.NewAggregate([]error{err, oErr})
+	}
+
+	return ctrl.Result{}, oErr
+}
+
+// updateAnnotations is for updating the annotations on the managed CRDs.
+func (r *Reconciler) updateAnnotations(crd *crdv1.CustomResourceDefinition) error {
+	annotations := crd.GetAnnotations()
+	if val, ok := annotations[common.CertManagerInjectCAFromAnnotation]; !ok || val != common.CertManagerInjectCAFromAnnotationValue {
+		patch := client.RawPatch(types.MergePatchType,
+			[]byte(fmt.Sprintf("{\"metadata\":{\"annotations\":{\"%s\":\"%s\"}}}",
+				common.CertManagerInjectCAFromAnnotation, common.CertManagerInjectCAFromAnnotationValue)),
+		)
+		if err := r.Patch(r.ctx, crd, patch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) updateAnnotationsInAllCRDs() error {
+	managedCRDList := &crdv1.CustomResourceDefinitionList{}
+	crdLabelFilter := map[string]string{
+		requestEnqueueLabelKey: requestEnqueueLabelValue,
+	}
+	if err := r.List(r.ctx, managedCRDList, client.MatchingLabels(crdLabelFilter)); err != nil {
+		return fmt.Errorf("failed to list managed CRD resources: %w", err)
+	}
```

Comment:
sorry, I read through this controller implementation a few times
but it is unclear to me which specific CRDs are being targetting at this step?

Reply:
initially, none of the CRDs will have `external-secrets.io/component: controller`, right? so will the controller upsert this label first..